sycl/include/sycl/ext/oneapi/device_global/device_global.hpp

```diff
@@ -44,7 +44,7 @@ template <typename T, typename PropertyListT, typename = void>
 class device_global_base {
 protected:
   using pointer_t = typename decorated_global_ptr<T>::pointer;
-  pointer_t usmptr{};
+  mutable pointer_t usmptr{};
```

Comment:
I think it would be useful to have a comment here that briefly mentions why this has to be mutable.